.gitlab-ci.yml

```diff
@@ -62,31 +53,26 @@ test:docker:
 
 publish:tests:
   stage: publish
-  image: golang:1.17-alpine3.14
+  image: python:3.11
   dependencies:
     - test
   before_script:
     # Install dependencies
-    - apk add --no-cache git
-    - GO111MODULE=off go get -u github.com/mattn/goveralls
-    # Coveralls env variables:
-    #  According to https://docs.coveralls.io/supported-ci-services
-    #  we should set CI_NAME, CI_BUILD_NUMBER, etc. But according
-    #  to goveralls source code (https://github.com/mattn/goveralls)
-    #  many of these are not supported. Set CI_BRANCH,
-    #  and pass few others as command line arguments.
-    #  See also https://docs.coveralls.io/api-reference
-    - export CI_BRANCH=${CI_COMMIT_BRANCH}
+    - apt update && apt install -yyq lcov
+    - pip install cpp-coveralls pyyaml
+
+    # eddyxu/cpp-coveralls appears dead, but there doesn't seem to be an
+    # alternative. Use this patch from someone who attempted to fix it. An
+    # alternative to this is to use pyyaml<6, but it's better to use just one
+    # old component than two.
```

Comment:
:+1:

Reply:
Shouldn't we apply the same to https://github.com/mendersoftware/mender-binary-delta/blob/master/.gitlab-ci.yml#L173 ?

Even better if we move this to a template. Our "unit tests" template assumes golang, but now we'll have two projects C/C++ so it makes sense to fork into a different template for them.

Reply:
It does make sense, but it makes testing it much harder, and I'm still not sure if this works completely the way it should (it's calling the branch "HEAD", and the mender-binary-delta version has some references to Travis variables... is it hardcoded in cpp-coveralls?). I think we still need to test this a bit in practice, so I think I'll create a task for it instead.

Reply:
https://tracker.mender.io/browse/QA-511